orttraining/orttraining/core/graph/gradient_schema_defs.cc

```diff
@@ -70,6 +70,32 @@ void AddRepeatedOutputs(
   }
 }
 
+static void propagateRecvOutputTensorElemTypes(
+    InferenceContext& ctx,
+    const std::string& attributeName,
+    size_t outputSize) {
+  auto attr_proto = ctx.getAttribute(attributeName);
+  if (nullptr == attr_proto) {  // attribute not present
+      fail_type_inference("Value of attribute ", attributeName, " not specified");
+  }
+
+  size_t tensor_num = static_cast<size_t>(attr_proto->ints_size());
+
+  if (tensor_num != outputSize) {
+    fail_type_inference("Attribute ", attributeName, " has a wrong size");
```

Comment:
User can't see which operator is wrong from this error message. Can we get the name from the context and print it here?